client/src/app/pages/identities/components/identity-form/kind-source-form.tsx

```diff
@@ -0,0 +1,188 @@
+  identity?: Identity;
+  defaultIdentities?: Record<IdentityKind, Identity | undefined>;
+}> = ({ identity, defaultIdentities }) => {
+  const { t } = useTranslation();
+  const { control, setValue, resetField } = useFormContext();
+  const values = useWatch({ control });
+
+  const [isKeyFileRejected, setIsKeyFileRejected] = useState(false);
+
+  const [isPasswordHidden, setIsPasswordHidden] = useState(true);
+  const toggleHidePassword = (e: React.FormEvent<HTMLElement>) => {
+    e.preventDefault();
+    e.stopPropagation();
+    setIsPasswordHidden(!isPasswordHidden);
+  };
+
+  const isPasswordEncrypted = identity?.password === values.password;
+  const isKeyEncrypted = identity?.key === values.key;
+  const kindDefault = defaultIdentities?.[values.kind as IdentityKind];
+  const isReplacingDefault =
+    values.default &&
+    kindDefault &&
+    (!identity || kindDefault.id !== identity.id);
+
+  return (
+    <>
+      <HookFormPFGroupController
+        control={control}
+        name="userCredentials"
+        label="User credentials"
+        isRequired
+        fieldId="user-credentials-select"
+        renderInput={({ field: { value, name, onChange } }) => (
+          <SimpleSelect
+            id="user-credentials-select"
+            toggleId="user-credentials-select-toggle"
+            toggleAriaLabel="User credentials select dropdown toggle"
+            aria-label={name}
+            value={
+              value ? toOptionLike(value, USER_CREDENTIALS_OPTIONS) : undefined
+            }
+            options={USER_CREDENTIALS_OPTIONS}
+            onChange={(selection) => {
+              const selectionValue =
+                selection as OptionWithValue<UserCredentials>;
+              onChange(selectionValue.value);
+              // So we don't retain the values from the wrong type of credential
+              resetField("default");
+              resetField("user");
+              resetField("key");
+              resetField("password");
+            }}
+          />
+        )}
+      />
+      <HookFormPFGroupController
+        control={control}
+        name="default"
+        fieldId="default"
+        label="Default credential?"
+        renderInput={({ field: { onChange, value, name, ref } }) => (
+          <>
+            <Switch
+              id="default"
+              name={name}
+              label={t("credentials.default.sourceSwitchLabel")}
+              isChecked={value}
+              onChange={(_, checked) => onChange(checked)}
+              ref={ref}
+            />
+            {isReplacingDefault && (
+              <Alert
+                isInline
+                className="alert-replacing-default"
+                variant="warning"
+                title={t("credentials.default.changeTitle")}
+              >
+                <Trans
+                  i18nKey="credentials.default.sourceChangeWarning"
+                  values={{
+                    name: kindDefault.name,
```

Comment:
`kindDefault` may be undefined

Reply:
Not according to typescript.  The `if (isReplacingDefault) {` removes the  undefined option from `kindDefault`